phases/14-agent-engineering/35-initialization-scripts/code/main.py

```
@@ -74,12 +106,101 @@ def probe_state_freshness() -> Probe:
+    """
+    if not LKG_PATH.exists():
+        return Probe("lkg_diff", "warn", "no last_known_good.json; pin one after first successful merge")
+    try:
+        lkg = json.loads(LKG_PATH.read_text())
+        baseline = lkg.get("commit")
+        if not baseline:
+            return Probe("lkg_diff", "warn", "lkg file present but commit field empty")
+    except json.JSONDecodeError as exc:
+        return Probe("lkg_diff", "fail", f"lkg file unreadable: {exc}")
+    try:
+        out = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)